pkg/util/cache_clean.go

```diff
@@ -0,0 +1,46 @@
+package util
+
+import "time"
+
+// CacheCleaner is an interface for cleaning a cache
+type CacheCleaner interface {
```

Comment:
Is there a reason why we need the `CacheCleaner` as well as the `IntervalCacheCleaner`? Are we going to remove the interval cache cleaner when the cleanable gets removed as part of the v2 cleanup?